flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/calls/ScalarOperatorGens.scala

```diff
@@ -191,12 +191,24 @@ object ScalarOperatorGens {
 
       case (TIMESTAMP_WITHOUT_TIME_ZONE, INTERVAL_DAY_TIME) =>
         generateOperatorIfNotNull(ctx, left.resultType, left, right) {
-          (l, r) => s"$l $op $r"
+          (l, r) => {
+            val leftTerm = s"$l.getMillisecond()"
+            val nanoTerm = s"$l.getNanoOfMillisecond()"
+            s"$SQL_TIMESTAMP.fromEpochMillis($leftTerm $op $r, $nanoTerm)"
```

Comment:
Do you think we can add some util to `SqlTimestamp`? like `plus(mills)` and etc.

Reply:
Not need now. Can add such things if we support more datetime arithmetic in the future.